File: 프로그래머스/lv2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
import java.io.*;
import java.util.*;

class Solution {
    public String solution(String s) {
        String[] split = s.toLowerCase().split("");
        // System.out.println(Arrays.toString(split));
        
        split[0] = split[0].toUpperCase();
        for(int i=1; i<split.length; i++) {
            if(split[i-1].equals(" ")) split[i] = split[i].toUpperCase();
        }
        
        String answer = "";
        for(int i=0; i<split.length; i++) {
            answer += split[i];
        }
        return answer;
    }
}

File: 프로그래머스/lv2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.js
function solution(s) {
    const split = s.toLowerCase().split('');
    
    split[0] = split[0].toUpperCase();
    let isWhiteSpace = false;
    const answer = [];
    for(let str of split) {
        str = isWhiteSpace ? str.toUpperCase() : str;
        isWhiteSpace = str == ' ' ? true : false;
        answer.push(str);
    }
    
    return answer.join('');
}